specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/examples/reservationGetByBillingAccount.json

```diff
@@ -0,0 +1,100 @@
+{
+  "parameters": {
+    "api-version": "2023-04-01",
+    "billingAccountName": "mock_billing_account",
```

Comment:
In BillingRP, we have the following pattern for billingAccountName:

```json
"billingAccountName": {
  "in": "path",
  "name": "billingAccountName",
  "description": "The ID that uniquely identifies a billing account.",
  "required": true,
  "type": "string",
  "pattern": "^([0-9]+|[0-9A-Fa-f]{8}-([0-9A-Fa-f]{4}-){3}[0-9A-Fa-f]{12}(:[0-9A-Fa-f]{8}-([0-9A-Fa-f]{4}-){3}[0-9A-Fa-f]{12}_[0-9]{4}(-[0-9]{2}){2})?)$",
  "x-ms-parameter-location": "method"
},
```

Once you define patterns for parameters, you samples will have to match the pattern to pass validations.

A modern billing account name that can pass the validation could be the following:

`00000000-0000-0000-0000-000000000000:00000000-0000-0000-0000-000000000000_2019-05-31`

Reply:
This pattern doesnt seem to exist in type.json file where `billingAccountName` is defined. I am going to create the definition separate definition of `billingAccountName` in reservation.json with this pattern

Reply:
@gaoyp830 - actually, can you take what types you need from our latest copy of [types.json](https://msazure.visualstudio.com/One/_git/ACE-BillingRP?path=/src/Documentation/swagger2.0/stable/2023-04-01/types.json&version=GBga/2023-04-01&_a=contents) and continue to use reference syntax instead of adding `billingAccountName` within reservations.json?